tests/test_precovery_index.py

```diff
@@ -1,88 +1,122 @@
+            )
+            orbits_keplerian.append(orbit)
+
+        # Load observations from h5 file
+        observations_df = pd.read_hdf(TEST_OBSERVATION_FILE)
+
+        # For each sample orbit, validate we get all the observations we planted
+        for orbit in orbits_keplerian:
+            results = precover(orbit, out_dir)
+
+            object_observations = observations_df[
+                observations_df["object_id"] == orbit_name_mapping[orbit.orbit_id]
+            ]
+            assert len(results) == len(object_observations) 
+            assert len(results) > 0
+
+            results.rename(
+                columns={
+                    "ra_deg": "ra",
+                    "dec_deg": "dec",
+                    "ra_sigma_arcsec": "ra_sigma",
+                    "dec_sigma_arcsec": "dec_sigma",
+                    "observation_id" : "obs_id",
+                    "obscode" : "observatory_code",
+                },
+                inplace=True,
+            )
+
+            results["ra_sigma"] /= 3600.0
+            results["dec_sigma"] /= 3600.0
+
+            # We are assuming that both the test observation file and the results
+            # are sorted by mjd_utc
+            for col in [
+                "mjd_utc",
+                "ra",
+                "ra_sigma",
+                "dec",
+                "dec_sigma",
+                "mag",
+                "mag_sigma",
+                # "filter", # can't do string comparisons this way
+            ]:
+                np.testing.assert_array_equal(
+                    object_observations[col].values, results[col].values
+                )
+
+            # Test that the observation_id, exposure_id, observatory_code, and filter
+            # are identical to the test observations
+            for col in [
+                "obs_id",
+                "exposure_id",
+                "observatory_code",
+                "filter",
+            ]:
+                assert (results[col].values == object_observations[col].values).all()
+
+            # Test that the predicted location of each objet in each exposure is 
+            # identical to the actual location of the object in that exposure (we did
+            # not add any errors to the test observations)
             np.testing.assert_array_equal(
-                object_observations[col].values, results[col].values
+                results[["pred_ra_deg", "pred_dec_deg"]].values,
+                object_observations[["ra", "dec"]].values,
             )
 
-    # TODO: validate length of frames table against what we would expect
+            # Test that the calculated distance is all zeros (again, we did not add
+            # any errors to the test observations)
+            np.testing.assert_array_equal(results["distance_arcsec"].values, np.zeros(len(results), dtype=np.float64))
+
+            
+        # TODO: validate length of frames table against what we would expect
+    
+    finally:
```

Comment:
It looks like `pytest` has a need fixtures setup / teardown API: https://docs.pytest.org/en/latest/how-to/fixtures.html#teardown-cleanup-aka-fixture-finalization

So we could do something like this:

```python
@pytest.fixture
def test_db_folder():
    out_dir = os.path.join(os.path.dirname(__file__), "database")
    yield out_dir
    shutil.rmtree(out_dir)
    
    
def test_precovery():
    index(
            out_dir=test_db_folder,
            dataset_id="test_dataset",
            dataset_name="Test Dataset",
            data_dir=os.path.join(os.path.dirname(__file__), "data"),
            ...
    )
```

That way we can re-use fixtures, preserve them in a programmatic way.

Reply:
Added in [4021523](https://github.com/B612-Asteroid-Institute/precovery/pull/51/commits/40215235b6f373ea39ceca6c2b39f2bd6e409c48)